src/FeedForwardPid.cpp

```diff
@@ -6,27 +6,45 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
+#include <limits>
+
 #include "iarc7_motion/FeedForwardPid.hpp"
 
 using namespace Iarc7Motion;
 
-FeedForwardPid::FeedForwardPid(double p_gain, double i_gain, double d_gain) : p_gain_(p_gain), i_gain_(i_gain), d_gain_(d_gain), setpoint_(0.0)
+FeedForwardPid::FeedForwardPid(double p_gain, double i_gain, double d_gain,
+                               double i_accumulator_max, double i_accumulator_min)
+    : p_gain_(p_gain),
+      i_gain_(i_gain),
+      d_gain_(d_gain),
+      setpoint_(0.0),
+      i_accumulator_(0.0),
+      last_current_value_(0.0),
+      i_accumulator_max_(i_accumulator_max),
+      i_accumulator_min_(i_accumulator_min)
 {
     // Nothing to do
 }
 
 double FeedForwardPid::update(double current_value, double time_delta)
 {
-
-    static double i_accumulator{0.0};
-    static double last_current_value{0.0};
-
+    if (time_delta == 0) {
+        time_delta = std::numeric_limits<double>::min();
```

Comment:
A p controller can do what you want to do here. A pid controller can not always accurately do it. There are control cases where what you want to do will not be accurate particularly if your timestep granularity is coarse.

I think the software is making a false promise if it allows for control values to be updated faster than a somewhat significant multiple of the clocks granularity. (In the case where the update rate is variable)

In the case that the update rate is fixed to a multiple of the clocks granularity. Input values should be sampled as close to the entry of the PID loop as possible.

In either case not doing so opens up your PID loop possible spikes in control output.